services/f-droid/f-droid.service.js

```diff
@@ -55,64 +64,31 @@ module.exports = class FDroid extends BaseYamlService {
     }
   }
 
-  async fetchYaml(url, options) {
-    const yaml = await this._requestYaml({
+  async fetch({ appId }) {
+    const url = `https://gitlab.com/kitsunyan/fdroidapi/raw/master/applications/${appId}`
+    return this._requestJson({
       schema,
-      url: `${url}.yml`,
-      ...options,
-    })
-    return { version: yaml.CurrentVersion }
-  }
-
-  async fetchText(url, options) {
-    const { buffer } = await this._request({
-      url: `${url}.txt`,
-      ...options,
-    })
-    const metadata = buffer.toString()
-    // we assume the layout as provided here:
-    // https://gitlab.com/fdroid/fdroiddata/raw/master/metadata/axp.tool.apkextractor.txt
-    const positionOfCurrentVersionAtEndOfTheFile = metadata.lastIndexOf(
-      'Current Version:'
-    ) // credits: https://stackoverflow.com/a/11134049
-    const lastVersion = metadata.substring(
-      positionOfCurrentVersionAtEndOfTheFile
-    )
-
-    const match = lastVersion.match(/^Current Version:\s*(.*?)\s*$/m)
-    if (!match) {
-      throw new InvalidResponse({
-        prettyMessage: 'invalid response',
-        underlyingError: new Error('could not find version on website'),
-      })
-    }
-    return { version: match[1] }
-  }
-
-  async handle({ appId }, { metadata_format: metadataFormat }) {
-    const url = `https://gitlab.com/fdroid/fdroiddata/raw/master/metadata/${appId}`
-    const fetchOpts = {
-      options: {},
+      url,
       errorMessages: {
+        403: 'app not found',
```

Comment:
When do we expect to get 403 response codes?

Reply:
Sometimes server may reply with 403 when API will be moved to f-droid.org. I can't imagine a "real" cases when it will reply this, but it's technically possible due to the nature of the API: it's not a real "REST" API, but instead it's just a bunch of statically generated files.

Reply:
I've got no strong opinion. If you want to keep the 403 case, let's leave it alone. :)